Spatial.gd

```diff
@@ -757,6 +749,97 @@ func clear_adjustment_nodes():
 		$Gizmos.remove_child(child)
 		child.queue_free()
 
+################################################################################
+# Update Waypoint datum
+################################################################################
+func set_icon_position(position: Vector3, icon_index: int):
+	var icon = self.currently_active_category.get_metadata(0).category3d.icons[icon_index]
```

Comment:
What if I have selected an icon not in whatever the currently active category is